domain/secret/service/watcher.go

```diff
@@ -270,7 +268,9 @@ func (w *secretWatcher[T]) loop() error {
 		if len(events) == 0 {
 			return nil
 		}
-		processed, err := w.processChanges(events.Values()...)
+		ctx, cancel := w.scopedContext()
+		processed, err := w.processChanges(ctx, events.Values()...)
```

Comment:
We typically create the ctx inside processChanges() and defer cancel() in that func
Can we stay consistent with the other places where this is implemented

Reply:
This is to keep it consistent with the secret backend package.
I am happy to change both the secret and secret backend implementation if you prefer that one.
But personally, I don't see much difference.